### PR TITLE
Add API for revocation and commands for OCSP generation/output

### PR DESCRIPTION
**Pull request by andrewbuss**
Rebased #458 onto master 

Add command and API endpoint to revoke a certificate in the database
Update ocspsign to accept reason strings
If a database is configured, add certificates signed locally to it 
Add ocsprefresh command: regenerate OCSP responses for unexpired certificates
Add ocspdump command: output previously generated OCSP responses in base64 format for ocspserve
Remove duplicate testing from test.sh

### CODE REVIEW

**Comment by andrewbuss** [comment]
I had to manually add goose to Godeps.json since it's only used in api/revoke/revoke_test. For some reason "go test" doesn't pull it in. Didn't catch this earlier since travis didn't run for the previous PR.

**Comment by lziest** [comment]
lgtm

File: cli/sign/sign.go [hunk 1]
```diff
@@ -56,13 +58,29 @@ func SignerFromConfig(c cli.Config) (signer.Signer, error) {
 	}
 
 	s, err := universal.NewSigner(cli.RootFromConfig(&c), policy)
+
```
Comment (lziest) [inline cli/sign/sign.go]: we can remove this blank line.

File: certdb/db_config.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+	cferr "github.com/cloudflare/cfssl/errors"
+	"github.com/cloudflare/cfssl/log"
+	_ "github.com/lib/pq"           // import just to initialize Postgres
+	_ "github.com/mattn/go-sqlite3" // import just to initialize SQLite
+)
+
+// DBConfig contains the database driver name and configuration to be passed to Open
+type DBConfig struct {
+	DriverName     string
```
Comment (lziest) [inline certdb/db_config.go]: nit: add json annotation to those fields, and make an example json that can be converted to DBConfig via LoadFile(), well yes I am asking for a test file for db_config. :)

File: cli/revoke/revoke.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
+// Package revoke implements the revoke command.
+package revoke
+
+import (
+	"database/sql"
+	goerr "errors"
```
Comment (lziest) [inline cli/revoke/revoke.go]: looks like we can remove the aliasing here

**Comment by lziest** [comment]
Thank you for the PR, appreciate it.

File: cli/revoke/revoke.go (hunk 3 above)
Comment (andrewbuss) [inline cli/revoke/revoke.go]: Removing the alias exposed golint nits about punctuation in error strings - looks like it only looks for `errors.New` and `fmt.Errorf`

**Comment by kisom** [comment]
LGTM.

**Pull request merged**